Sources/MapboxNavigation/FloatingButton.swift

```diff
@@ -44,16 +44,30 @@ open class FloatingButton: Button {
      - parameter image: The `UIImage` of this button.
      - parameter selectedImage: The `UIImage` of this button when selected.
      - parameter size: The size of this button,  or `FloatingButton.buttonSize` if this argument is not specified.
+     - parameter type: `UIButton` type. Defaults to `.custom`.
+     - parameter cornerRadius: Corner radius of the button.
+     
+     - returns: `FloatingButton` instance.
      */
     public class func rounded<T: FloatingButton>(image: UIImage? = nil,
                                                  selectedImage: UIImage? = nil,
-                                                 size: CGSize = FloatingButton.buttonSize) -> T {
-        let button = T.init(type: .custom)
+                                                 size: CGSize = FloatingButton.buttonSize,
+                                                 type: UIButton.ButtonType = .custom,
+                                                 cornerRadius: CGFloat = FloatingButton.buttonSize.width / 2.0) -> T {
+        let button = T.init(type: type)
         button.translatesAutoresizingMaskIntoConstraints = false
         button.constrainedSize = size
-        button.setImage(image, for: .normal)
-        if let selected = selectedImage { button.setImage(selected, for: .selected) }
-        button.applyDefaultCornerRadiusShadow(cornerRadius: size.width / 2)
```

Comment:
This is a tradeoff for performance as noted above. I think it could look like an intentional design change, as opposed to a bug – not dissimilar to changes in Apple software over the years. But the main thing is to make sure the border is sufficiently discernible.

Reply:
I think two major reasons for this change is consistency and only partly performance. Previously we've shown shadow, border or no border in views where consistency is expected (top and bottom banners, resume button, floating button). With shadow removal we make look more consistent and customizable to bring new UI updates for next iteration of drop-in.